### DISCUSSION TITLE
HL Commander with Flowdeck

### DISCUSSION THREAD

**Discussion by puliceworx** [bitcraze/discussions#806] [Q&A] [answered]
Greetings, I just completed watching the new presentation (Crazyflie 101 EPFL-LIS Lecture 2023). 

When discussing the commanders it was stated that High-Level Commander is not advisable for use with a flowdeck and should use a positioning system.  

I wonder if that can be expanded upon?  Is it because of how the hl commander is coded in firmware? Accuracy?  

I understand motion commander and hl commander are both based on velocity, so the difference would be absolute vs relative position within the cf firmware? 

Regards and thank you.

**Comment by knmcguire** [comment, marked as answer]
Hi!

Happy you liked the presentation!

It's more in the sense of, that all the functions of the highlevel commander from the cflib are made with the afterthought that you have a global positioning system. None the less, taking off would probably work fine due that you have an accurate height estimate, but in the high level commander there is no function implemented that says: 'move with 0.3 m/s forward'. It is always 'move to 2 meters to the front with a speed of 0.3 m/s'.  So there is always a position element in there, and with the flowdeck you'll have potential drift with it as well, so it might not be great.

But it most likely will work anyway, that is no problem, but it might not perform as expected. For anything with flow based autonomy, you might just only want to control the speed in the horizontal plane and make it reactively stop by sensing something with the multiranger for instance.

So long story short, it's more about what kind of functions are implemented. You can check that in the API documentation of both:

- https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/crazyflie/high_level_commander/
- https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/positioning/motion_commander/
